.tekton/stonesoupe-e2e-test.yaml

```diff
@@ -0,0 +1,249 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: stonesoupe-e2e-test
+  annotations:
+    pipelinesascode.tekton.dev/on-event: "[pull_request, push]"
+    pipelinesascode.tekton.dev/on-target-branch: "main"
+    pipelinesascode.tekton.dev/task: "[git-clone]"
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
+spec:
+  params:
+    - name: repo_url
+      value: "{{ repo_url }}"
+    - name: revision
+      value: "{{ revision }}"
+    - name: repo_owner
+      value: "{{ repo_owner }}"
+  timeouts:
+    pipeline: "2h0m0s"
+    finally: "0h20m0s"
+  pipelineSpec:
+    params:
+      - name: repo_url
+      - name: revision
+      - name: repo_owner
+    workspaces:
+      - name: source
+      - name: kubeconfig-dir
+      - name: shared-workspace
+    tasks:
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+          kind: ClusterTask
+        workspaces:
+          - name: output
+            workspace: source
+        params:
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+      - name: produce-cluster-name
+        runAfter:
+          - "fetch-repository"
+        taskSpec:
+          results:
+            - name: cluster-name
+              description: Openshift cluster name
+          steps:
+            - name: cluster-name
+              image: registry.access.redhat.com/ubi9/openssl:9.1-2
+              script: |
+                #!/usr/bin/env bash
+                set -o errexit
+                set -o nounset
+                set -o pipefail
+                set -x
+                CLUSTER_NAME="ci-$( openssl rand -hex 5 )"
+                echo -n "$CLUSTER_NAME" | tee $(results.cluster-name.path)
+      - name: deploy-cluster
```

Comment:
I think there is lot many common tasks we are using in our pipelineRun. Can we separate out the task steps in other file, for ex: `.tekton/tasks/deploy-cluster` and just use it inside the pipelineRun

Reply:
> I think there is lot many common tasks we are using in our pipelineRun. Can we separate out the task steps in other file, for ex: `.tekton/tasks/deploy-cluster` and just use it inside the pipelineRun

good idea. I will open a PR to refactor them after this PR is merged